hera_sim/simulate.py

```diff
@@ -31,226 +39,386 @@ def new_func(*args, **kwargs):
     return new_func
 
 
-# FIXME: some of the code in here is pretty brittle and breaks (sometimes silently)
-# if not used carefully. This definitely needs a review and cleanup.
 class Simulator:
     """Class for managing a simulation.
 
     """
 
-    def __init__(self, data=None, defaults_config=None, **kwargs):
-        """Initialize a Simulator object.
-
-        Idea: Make Simulator object have three major components:
-            sim.data -> UVData object for storing the "measured" data
-                Also keep track of most metadata here
-            sim.defaults -> Defaults object
-
+    def __init__(
+        self,
+        *,
+        data: Optional[Union[str, UVData]] = None,
+        defaults_config: Optional[Union[str, Dict]] = None,
+        redundancy_tol: float = 1.0,
+        **kwargs,
+    ):
+        """Simulate visibilities and instrumental effects for an entire array.
```

Comment:
In the `update-docs` branch I decided to go with putting the class docstring at the class level instead of in `__init__`, and turned off checks for having docs in `__init__`. I had a good look around the web to check what was best practice and tbh no-one really knows. But the numpydoc standard seems to indicate they should be put in the class docs.

Reply:
I moved the `__init__` docstring to the class docstring for the `Simulator`. Not sure if we should have anything in the `__init__` docstring, so it's currently empty.